Add some new use cases to Traffic Policy examples (#1169)
ngrok/ngrok-docs: docs/traffic-policy/examples/index.mdx

A number of these examples come from a longer article about how ngrok [makes policy management accessible](https://ngrok.com/blog-post/api-gateway-policy-management-examples) to developers, including a simple Go-based application for testing these and other configurations.

See the following categories for specific expressions and actions:
## Route requests

- [Authentication](#authentication)
- [Rate limiting](#rate-limiting)
- [Block unwanted requests](#block-unwanted-requests)

> - [Other](#other)

Reviewer: These links were broken, but I just removed this content entirely because all it does is recreate the TOC. In general I frown upon that because now there are two TOCs to maintain on the page, and if the links are ever broken it looks unprofessional

> interpolation](/docs/traffic-policy/concepts/cel-interpolation.mdx) to

Reviewer: ```suggestion
interpolation](/docs/traffic-policy/concepts/cel-interpolation) to
```
We shouldn't add `.mdx` to the ends of these links

hostname, path, headers, and more. This allows you to manage complex traffic
topologies without manually creating and managing each [`forward-internal`
action](/docs/traffic-policy/actions/forward-internal.mdx).

Forward requests from `https://*.example.com` to `https://<SUBDOMAIN>.internal`.

```yaml mode=traffic-policy
on_http_request:
  - actions:
      - type: forward-internal
        config:

>           url: https://${req.host.split('.example.com)[0]}.internal

Reviewer: Would this actually work, considering the single apostrophe at `'.example`?
`req.host.split('.example.com)[0]`

```

Forward requsets containing a `X-Customer-Value: {CUSTOMER}` header to `https://{CUSTOMER}.internal`.

```yaml mode=traffic-policy
on_http_request:
  - actions:
      - type: "forward-internal"
        config:

>           url: https://${getReqHeader(‘X-Customer-Value’)[0]}.internal

Reviewer: ```suggestion
          url: https://${getReqHeader('X-Customer-Value')[0]}.internal
```
The single quotes looked weird for some reason

<LimitSize />

## Manipulate headers

### Enrich your upstream service

Add new headers to requests to give your upstream service more context about the
consumer, which in turn allows for richer functionality, such as localized
languages and pricing.

```yaml mode=traffic-policy
on_http_request:
  - actions:
      - type: add-headers
        config:
          headers:
						x-is-ngrok: "1"
            x-endpoint-id: "${endpoint.id}"
            x-client-ip: "${conn.client_ip}"
            x-client-conn-start: "${conn.ts.start}"
            x-client-loc: "${conn.geo.city}, ${conn.geo.country}"
            x-client-path: "${req.url.path}
```

### Remove service details from response headers

> Some frameworks, like Express, add headers like `X-Powered-By: Express` to

Reviewer: ```suggestion
Some frameworks, like [Express](https://expressjs.com/), add headers like `X-Powered-By: Express` to
```

> responses, which you may not want to reveal to your users.

Reviewer: ```suggestion
responses, which you may not want to reveal to your users.

The following example removes the `X-Powered-By` header.
```

        config:
          headers:
            - "X-Powered-By"
```

## Other

### Deploy simple A/B tests

Using the [`rand.double()`
macro](/docs/traffic-policy/macros/utility.mdx#randdouble--double) set to `>=
0.5`, you can equally split incoming requests to two different internal agent
endpoints, which forward traffic to the two versions of your service.

You can manipulate `<= 0.5` to match the percentage of requests to route to your
B test, e.g. `<= 0.1` for 10%.

```yaml mode=traffic-policy
---
on_http_request:
  - expressions:
      - "rand.double() <= 0.5"
    actions:
      - type: "forward-internal"
        config:
          url: https://b.internal
  - actions:
      - type: "forward-internal"
        config:
          url: https://a.internal
```

If you don't have multiple services, you could also route B traffic to a
separate route.

```yaml mode=traffic-policy
---
on_http_request:
  - expressions:
      - "rand.double() <= 0.5"
    actions:
      - type: "url-rewrite"
        config:
					from: "/path/to/test"
          to: "/path/to/test-b"
```

### Create 'pretty' URLs for SEO

You can map the permalinks created by a blog CMS to "pretty" alternatives that

> are easier for both humans and SEO bot to grok. The following rule rewrites the

Reviewer: ```suggestion
are easier for both humans and [SEO bots](https://developers.google.com/search/docs/crawling-indexing/googlebot) to understand.

The following rule rewrites the
```
Appreciate the use of grok 🤣, but not everyone knows what it means
